lib/datadog/tracing/span.rb

```diff
@@ -204,6 +204,13 @@ def pretty_print(q)
         end
       end
 
+      # Single Span Sampling has chosen to keep this span
+      # regardless of the trace-level sampling decision
+      # @!visibility private
+      def single_sampled?
```

Comment:
I don't feel like this responsibility is quite right. `Span` should only be describing its own state, but now it seems to know something about the nature of span sampling. I don't like the presence of sampling logic and references in `Span` because I don't think it's a span's responsibility to know how it's sampled. It's supposed to be dumb.

It may be better to extract out this `single_sampled?` logic elsewhere, presumably to the thing that is aware of sampling that uses this. If this isn't possible/practical, I think at minimum this should be reduced to a simple variable e.g. `@single_sampled`.